src/PowerShellEditorServices/Services/DebugAdapter/Debugging/BreakpointDetails.cs

```diff
@@ -55,7 +57,8 @@ private BreakpointDetails()
             int line,
             int? column = null,
             string condition = null,
-            string hitCondition = null)
+            string hitCondition = null,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4598462)

Reply:
It's flagging these just because they're public APIs

Reply:
> It's flagging these just because they're public APIs

Yeah, optional parameters are kind of a pain in public API's.  Overloads are typically better because their value is a constant in the callers CIL.  So if you change the default value and ship a new version, folks need to recompile before they'll get the new default value.

It also adding a new parameter with a default is a binary breaking change.  It doesn't actually create a new overload behind the scenes, so old code is still trying to reference the method with the old signature, which will throw at JIT time.

Also some CLR languages don't support it apparently.